### PR TITLE
Update ADAM to Python 3.7

### PR DESCRIPTION
**Pull request by lichtefeld**
This PR updates ADAM to use Python 3.7 now that 3.6 is end of life.

### CODE REVIEW

File: tests/language_specific/english/test_english_language_generator.py [hunk 1]
```diff
@@ -1148,108 +1148,90 @@ def test_in_region_as_goal():
...
+            goal_object,
         )
-        == ("a", "dog", "goes", "over", "a", "table")
-    )
+    ) == ("a", "dog", "goes", "over", "a", "table")
 
 
 def test_under_region_as_goal():
     goal_object = situation_object(TABLE)
     # Over
```
Comment (danielnapierski) [inline tests/language_specific/english/test_english_language_generator.py:1228]: 1243 better as "Under"

File: tests/language_specific/chinese/test_chinese_language_generator.py [hunk 2]
```diff
@@ -2202,94 +2200,79 @@ def test_in_region_as_goal():
...
 def test_under_region_as_goal():
     goal_object = situation_object(TABLE)
     # Over
-    assert (
-        generated_tokens(
-            region_as_goal_situation(
-                Region(goal_object, distance=PROXIMAL, direction=GRAVITATIONAL_DOWN),
-                goal_object,
-            )
+    assert generated_tokens(
```
Comment (danielnapierski) [inline tests/language_specific/chinese/test_chinese_language_generator.py:2270]: line 2269 "# Under"

File: requirements_lock.txt [hunk 3]
```diff
@@ -1,63 +1,60 @@
...
-charset-normalizer==2.0.6
-click==8.0.1
+black==22.1.0
+certifi==2021.10.8
+charset-normalizer==2.0.11
+click==8.0.3
 click-default-group==1.2.2
 contexttimer==0.3.3
-coverage==6.0.1
-dataclasses==0.8
```
Comment (danielnapierski) [inline requirements_lock.txt:12]: was datac;asses unused?

File: Makefile [hunk 4]
```diff
@@ -2,7 +2,7 @@ default:
 	@echo "an explicit target is required"
 
 # easier to test python2 vs. python3
-PYTHON=python3
+PYTHON=pypy3
```
Comment (danielnapierski) [inline Makefile:5]: Can you comment on the benefits of this change?

**Review by danielnapierski — APPROVED** [review]
Looks good.  Added a couple of questions.

File: Makefile (hunk 4 above)
Comment (lichtefeld) [reply Makefile:5]: We were previously using pypy for tests as it runs the entire test suite in ~10ish minutes on my laptop as compared to ~20+ for regular python. A large number of the tests involve our graph matching algorithm, which while designed to be efficient, is solving an NP-Complete problem of sub-graph isomorphism matching. I'd run into problems getting the local install of pypy working so I had changed this back to python3 so I could still test even if slower but with the update to 3.7 I got pypy working again locally.

File: requirements_lock.txt (hunk 3 above)
Comment (lichtefeld) [reply requirements_lock.txt:12]: dataclasses are built-in to python 3.7 so the backwards compatibility package can be dropped.

**Pull request merged**